.github/workflows/frontend.yml

```diff
@@ -81,26 +85,68 @@ jobs:
           key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
           restore-keys: |
             ${{ runner.os }}-yarn-
+      - name: go mod cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-mod-${{ hashFiles('**/go.sum') }}  
+          restore-keys: |
+            ${{ runner.os }}-go-mod-
+      - name: go build cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-build-
       - name: Install dependencies
         if: steps.cache.outputs.cache-hit != 'true'
         run: yarn --cwd frontend install
       - name: Compile packages
-        run: yarn --cwd frontend compile
-      - name: Test [unit]
-        run: yarn --cwd frontend test:coverage
+        run: yarn --cwd frontend compile:dev
       - name: Test [e2e]
         id: e2e
-        run: make frontend-e2e
+        run: ./tools/frontend-e2e.sh
       - name: Upload end-to-end artifacts
         uses: actions/upload-artifact@v2
         if: failure() && steps.e2e.outcome == 'failure'
         with:
           name: screenshots
           path: ./frontend/packages/app/cypress/screenshots
+  test-unit:
+    name: test unit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+          check-latest: true
+      - name: Enforce consistent Yarn version
+        run: ./tools/install-yarn.sh
+      - name: Get yarn cache
+        id: yarn-cache
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v2
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
```

Comment:
I'm not sure how this fully works but should this be `steps.yarn-cache`? Are the keys in this statement here the `id` of each step?

Reply:
oh yeah good catch